src/compiler/checker.ts

```diff
@@ -674,6 +676,12 @@ namespace ts {
             return result;
         }
 
+        function doesNameResolveToLocalSymbolForExportDefault(exports: SymbolTable, meaning: SymbolFlags, name: string): boolean {

```

Comment:
Yes, that's more readable. (Vlad and Ron pointed me to the binder too, but I couldn't get the changes to work out, so I ended up back here in the checker.)